src/mail-app/mail/view/MailView.ts

```diff
@@ -500,12 +500,14 @@ export class MailView extends BaseTopLevelView implements TopLevelView<MailViewA
 	}
 
 	private getSetUnreadStateAction(): ((unread: boolean) => void) | null {
-		const mails = this.mailViewModel.listModel?.getSelectedAsArray() ?? []
-		if (isEmpty(mails) || (mails.length === 1 && isDraft(getFirstOrThrow(mails)))) {
+		const conversationMails = this.conversationViewModel?.conversationMails()
```

Comment:
it gets pretty complicated, is there any way we can shove it into ViewModel? same goes for other changes in this file